hive/src/main/java/com/netease/arctic/hive/utils/HiveMetaSynchronizer.java

```diff
@@ -0,0 +1,222 @@
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Synchronize the metadata of the hive table to arctic table
+ */
+public class HiveMetaSynchronizer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HiveMetaSynchronizer.class);
+
+  /**
+   * Synchronize the schema change of the hive table to arctic table
+   * @param table arctic table to accept the schema change
+   * @param hiveClient hive client
+   */
+  public static void syncHiveSchemaToArctic(ArcticTable table, HMSClient hiveClient) {
+    try {
+      Table hiveTable = hiveClient.run(client -> client.getTable(table.id().getDatabase(), table.id().getTableName()));
+      List<FieldSchema> fieldSchemas =  hiveTable.getSd().getCols();
+      Schema hiveSchema = org.apache.iceberg.hive.HiveSchemaUtil.convert(fieldSchemas);
+      UpdateSchema updateSchema = table.updateSchema();
+      boolean update = false;
+      for (int i = 0; i < hiveSchema.columns().size(); i++) {
+        Types.NestedField hiveField = hiveSchema.columns().get(i);
+        Types.NestedField icebergField = table.schema().findField(hiveField.name());
+        if (icebergField == null) {
+          updateSchema.addColumn(hiveField.name(), hiveField.type(), hiveField.doc());
+          update = true;
+          LOG.info("Table {} sync new hive column {} to arctic", table.id(), hiveField);
+        } else if (!icebergField.type().equals(hiveField.type()) ||
+            !Objects.equals(icebergField.doc(), (hiveField.doc()))) {
+          updateSchema.updateColumn(icebergField.name(), icebergField.type().asPrimitiveType(), hiveField.doc());
+          update = true;
+          LOG.info("Table {} sync updated hive column {} to arctic", table.id(), icebergField);
+        }
+      }
+      if (update) {
+        updateSchema.commit();
+      }
+    } catch (TException | InterruptedException e) {
+      throw new RuntimeException("Failed to get hive table:" + table.id(), e);
+    }
+  }
+
+  /**
+   * Synchronize the data change of the hive table to arctic table
+   * @param table arctic table to accept the data change
+   * @param hiveClient hive client
+   */
+  public static void syncHiveDataToArctic(ArcticTable table, HMSClient hiveClient) {
+    UnkeyedTable baseStore;
+    if (table.isKeyedTable()) {
+      baseStore = table.asKeyedTable().baseTable();
+    } else {
+      baseStore = table.asUnkeyedTable();
+    }
+    try {
+      if (table.spec().isUnpartitioned()) {
+        Table hiveTable =
+            hiveClient.run(client -> client.getTable(table.id().getDatabase(), table.id().getTableName()));
+        String hiveTransientTime =  hiveTable.getParameters().get("transient_lastDdlTime");
+        String arcticTransientTime = baseStore.partitionProperty().get(TablePropertyUtil.EMPTY_STRUCT)
+            .get(HiveTableProperties.PARTITION_PROPERTIES_KEY_TRANSIENT_TIME);
+        if (arcticTransientTime == null || !arcticTransientTime.equals(hiveTransientTime)) {
```

Comment:
The logic is hiveTransientTime  > arcticTransientTime  for sync hive data to arctic?
Because if hiveTransientTime  < arcticTransientTime, is sync arctic data to hive

Reply:
transient time is not trustable so i would not compare its value.
Does ams would compare this value when sync arctic data to hive?

Reply:
> transient time is not trustable so i would not compare its value. Does ams would compare this value when sync arctic data to hive?

No, ams wouldn't compare this value.Because only arctic will change the partition location, once ams found the arctic partition location is different with hive partition, ams will sync arctic data to hive.
OK, I understand your mean. According your logic, the conditions should add check arctic partition location whether the same as  hive partition.If not, should sync arctic data to hive.Rather only compare TransientTime?

Reply:
> > transient time is not trustable so i would not compare its value. Does ams would compare this value when sync arctic data to hive?
> 
> No, ams wouldn't compare this value.Because only arctic will change the partition location, once ams found the arctic partition location is different with hive partition, ams will sync arctic data to hive. OK, I understand your mean. According your logic, the conditions should add check arctic partition location whether the same as hive partition.If not, ams should sync arctic data to hive. Rather only compare TransientTime?